src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -640,6 +702,43 @@ static void G(A a, B b)
             Assert.Equal("I<(System.Object X, System.Object? Y)>", type.Interfaces()[0].ToTestDisplayString());
         }
 
+        [Fact]
+        public void EmitAttribute_ImplementedInterfaces()
+        {
+            var source =
+@"#nullable enable
+public interface I<T> { }
+public class A :
+    I<object>,
+    I<string?>
+{
+    public object FA1;
+    public object FA2;
+}
+public class B :
+#nullable disable
+    I<object>,
+#nullable enable
+    I<int>
+{
+    public object FB1;
+    public object FB2;
+}";
+            var comp = CreateCompilation(source);
+            var expected =
+@"[NullableContext(2)] I<T>
+    T
+[NullableContext(1)] [Nullable(0)] A
```

Comment:
>[NullableContext(1)]  [](start = 0, length = 21)

I am not sure if this test sufficiently asserts that the interfaces are taken into account. Would it make sense to add a scenario where interfaces are dominating the decision about the context?

Reply:
`B` demonstrates the effect of interfaces.

---
In reply to: [297226952](https://github.com/dotnet/roslyn/pull/36152#discussion_r297226952) [](ancestors = 297226952)